File: src/SummaryRanges.java
import java.util.ArrayList;
import java.util.List;

public class SummaryRanges {

    private String getRangeString(int start, int end) {
        return end == start ? start + "" : start + "->" + end;
    }

    public List<String> summaryRanges(int[] array) {
        if (array.length == 0) return new ArrayList<>();
        List<String> result = new ArrayList<>();
        int startIndex = 0;
        for (int index = 1 ; index < array.length ; index++) {
            if (array[index] > array[startIndex] + (index - startIndex)) {
                result.add(getRangeString(array[startIndex], array[index - 1]));
                startIndex = index;
            }
        }
        result.add(getRangeString(array[startIndex], array[array.length - 1]));
        return result;
    }
}

File: python/summary_ranges.py
from typing import List


class Solution:
    def getRangeString(self, start: int, end: int) -> str:
        return f'{start}' if start == end else f'{start}->{end}'

    def summaryRanges(self, array: List[int]) -> List[str]:
        if len(array) == 0: return []
        result = []
        startIndex, index = 0, 1
        while index < len(array):
            if array[index] > array[startIndex] + (index - startIndex):
                result.append(self.getRangeString(array[startIndex], array[index - 1]))
                startIndex = index
            index += 1
        result.append(self.getRangeString(array[startIndex], array[len(array) - 1]))
        return result
